pkg/cmd/admin/top/graph.go

```diff
@@ -0,0 +1,149 @@
+				topLayerAdded = true
+			}
+			g.AddEdge(imageNode, layerNode, edgeKind)
+			glog.V(4).Infof("Adding image layer %q to graph (%q) ", layer.Name, edgeKind)
+		}
+	}
+}
+
+func addImageStreamsToGraph(g graph.Graph, streams *imageapi.ImageStreamList) {
+	for i := range streams.Items {
+		stream := &streams.Items[i]
+		glog.V(4).Infof("Adding ImageStream %s/%s to graph", stream.Namespace, stream.Name)
+		isNode := imagegraph.EnsureImageStreamNode(g, stream)
+		imageStreamNode := isNode.(*imagegraph.ImageStreamNode)
+
+		// connect IS with underlying images
+		for tag, history := range stream.Status.Tags {
+			for i := range history.Items {
+				image := history.Items[i]
+				n := imagegraph.FindImage(g, image.Image)
+				if n == nil {
+					glog.V(2).Infof("Unable to find image %q in graph (from tag=%q, dockerImageReference=%s)",
+						history.Items[i].Image, tag, image.DockerImageReference)
+					continue
+				}
+				imageNode := n.(*imagegraph.ImageNode)
+				glog.V(4).Infof("Adding edge from %q to %q", imageStreamNode.UniqueName(), imageNode.UniqueName())
+				edgeKind := ImageStreamEdgeKind
+				if i > 1 {
+					edgeKind = HistoricImageStreamEdgeKind
+				}
+				g.AddEdge(imageStreamNode, imageNode, edgeKind)
+			}
+		}
+	}
+}
+
+func addPodsToGraph(g graph.Graph, pods *kapi.PodList) {
+	for i := range pods.Items {
+		pod := &pods.Items[i]
+		if pod.Status.Phase != kapi.PodRunning && pod.Status.Phase != kapi.PodPending {
+			glog.V(4).Infof("Pod %s/%s is not running nor pending - skipping", pod.Namespace, pod.Name)
+			continue
+		}
+
+		glog.V(4).Infof("Adding pod %s/%s to graph", pod.Namespace, pod.Name)
+		podNode := kubegraph.EnsurePodNode(g, pod)
+		addPodSpecToGraph(g, &pod.Spec, podNode)
+	}
+}
+
+func addPodSpecToGraph(g graph.Graph, spec *kapi.PodSpec, predecessor gonum.Node) {
+	for j := range spec.Containers {
+		container := spec.Containers[j]
+
+		glog.V(4).Infof("Examining container image %q", container.Image)
+		ref, err := imageapi.ParseDockerImageReference(container.Image)
+		if err != nil {
+			glog.V(2).Infof("Unable to parse DockerImageReference %q: %v - skipping", container.Image, err)
+			continue
+		}
+
+		if len(ref.ID) == 0 {
+			// ignore not managed images
+			continue
+		}
+
+		imageNode := imagegraph.FindImage(g, ref.ID)
+		if imageNode == nil {
+			glog.V(1).Infof("Unable to find image %q in the graph", ref.ID)
+			continue
+		}
+
+		glog.V(4).Infof("Adding edge from %v to %v", predecessor, imageNode)
+		g.AddEdge(predecessor, imageNode, PodImageEdgeKind)
+	}
+}
+
+func markParentsInGraph(g graph.Graph) {
+	imageNodes := getImageNodes(g.Nodes())
+	for _, in := range imageNodes {
+		// find image's top layer, should be just one
+		for _, e := range g.OutboundEdges(in, ImageTopLayerEdgeKind) {
+			layerNode, _ := e.To().(*imagegraph.ImageLayerNode)
+			// find image's containing this layer but not being their top layer
+			for _, ed := range g.InboundEdges(layerNode, ImageLayerEdgeKind) {
```

Comment:
To demonstrate, here's an image built using Dockerfile:

```sh
$ cat Dockerfile
127.0.0.1:5002/miminar/busybox
COPY data /data

$ cat blobs/sha256/00/0057af9ede993a9fb64d86e1392103e8282ec03434e95ceea5f46b4b6f1e3494/data 
{
   "schemaVersion": 1,
   "name": "miminar/data",
   "tag": "buxybox-based",
   "architecture": "amd64",
   "fsLayers": [
      {
         "blobSum": "sha256:0ec44ea3d1c16c2efa1377e55d973b5ef2bbd53cf8b309e5f13fa296bca9d658"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:eeee0535bf3cec7a24bff2c6e97481afa3d37e2cdeff277c57cb5cbdb2fa9e92"
      }
   ],
```

And here's another image based on different image with the same copy line:

```sh
cat Dockerfile
FROM miminar/hwdata-check
COPY data /data

$ cat blobs/sha256/15/156ec81a6e3a00eaa092caf4a4e9b5f24df20bb1fae1622acc85f22beefd86c6/data 
{
   "schemaVersion": 1,
   "name": "miminar/data",
   "tag": "hwdata-check-based",
   "architecture": "amd64",
   "fsLayers": [
      {
         "blobSum": "sha256:0ec44ea3d1c16c2efa1377e55d973b5ef2bbd53cf8b309e5f13fa296bca9d658"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:d27122453b78d8b81fb83247b136b02df4d45b8d49273d42263ee5ce8411d110"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:4d92024ca18c90fc310efff77203062188044de6a492b464ba2efc5ce7adbde7"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      },
      {
         "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
      }
   ],
```

Note the same digest of the top layer and completely different blob history.